kong/dao/cassandra_db.lua

```diff
@@ -69,8 +69,16 @@ local function deserialize_rows(rows, schema)
   local json = require "cjson"
   for i, row in ipairs(rows) do
     for col, value in pairs(row) do
-      if schema.fields[col].type == "table" then
-        rows[i][col] = json.decode(value)
+      if schema.fields[col].type == "table" or schema.fields[col].type == "array" then
+        local success, result = pcall(json.decode, value);
+        if success then
+          rows[i][col] = result
+        else
```

Comment:
Quick comment about this: the aim of this code is to allow the code to retrieve oauth2 credentials whose redirect_uri were saved before this PR is (hopefully) merged. For legacy records, redirect_uri is indeed not serialized, it's just a string.

I hesitated between doing so and writing a migration file that would convert all the redirect_uri in the new array format. I didn't do so because:
- I believe this approach is better to be able to upgrade a cluster of kong nodes without downtime,
- I don't feel comfortable coding migration files, and don't know how to test them.

If you guys would rather have a migration file, I would appreciate you help me on that one.